engine/language-server/src/main/scala/org/enso/languageserver/protocol/json/JsonRpc.scala

```diff
@@ -29,12 +29,14 @@ import org.enso.languageserver.workspace.WorkspaceApi.ProjectInfo
 
 object JsonRpc {
 
-  /** A description of supported JSON RPC messages.
-    */
-  val protocol: Protocol = Protocol.empty
-    .registerRequest(Ping)
+  /** A description of JSON RPC messages support during the initialization stage */
+  val initProtocol: Protocol = Protocol.empty
     .registerRequest(InitialPing)
     .registerRequest(InitProtocolConnection)
+
+  /** A description of supported JSON RPC messages at a post-initialization stage */
+  def fullProtocol(init: Protocol): Protocol = init
+    .registerRequest(Ping)
```

Comment:
The language server should be able to receive `InitProtocolConnection` after the initialization too https://github.com/enso-org/enso/blob/e7a0312298e212ee5c3ed58804eec44022515566/engine/language-server/src/main/scala/org/enso/languageserver/protocol/json/JsonConnectionController.scala#L283-L289

Reply:
And it will, since it is already part of `init` argument.